core/src/testils/criterion.rs

```diff
@@ -40,4 +49,47 @@ impl BencherExt for Bencher<'_> {
             total_elapsed
         })
     }
+
+    /// Similar to criterion's `iter_batched`, but uses a proptest strategy,
+    /// that returns an mbuf, as the setup to randomly generate a batch
+    /// inputs that can be polled for benchmarking pipeline-combinations and
+    /// operators.
+    fn iter_proptest_polled<R, S: Strategy<Value = Mbuf>, O: Batch>(
+        &mut self,
+        strategy: S,
+        mut routine: R,
+        batch_size: usize,
+    ) where
+        R: FnMut(Poll<Receiver<Mbuf>>) -> O,
+    {
+        self.iter_custom(|mut iters| {
+            let mut total_elapsed = Duration::from_secs(0);
+            let mut gen = Rvg::deterministic();
+            while iters > 0 {
+                let batch_size = cmp::min(batch_size, iters as usize);
+                let inputs = black_box(gen.generate_vec(&strategy, batch_size));
+                let mut outputs = Vec::with_capacity(batch_size);
+
+                let (mut tx, rx) = mpsc::channel();
```

Comment:
couldn't get poll_fn to work?

Reply:
Nope. Not with the tote needs; that closure issue.

Reply:
might work if the input to routine is also generically type.

```
 R: FnMut(I) -> O
```

and `I: Batch`.

Reply:
@drunkirishcoder this is what I tried, for poll_fn and ran into the issue of involving a closure it couldn't infer or needing a separate version for using `dyn`.

Reply:
ok